discord.d.ts

```diff
@@ -663,10 +663,10 @@ declare module "discord.js" {
         deleteAll(): Promise<void[]>;
         every(fn: Function, thisArg?: Object): boolean;
         exists(prop: string, value: any): boolean;
-        filter(fn: Function, thisArg?: Object): Collection<any, any>;
-        find(propOrFn: string | Function, value?: any): any;
-        findAll(prop: string, value: any): any[];
-        findKey(propOrFn: string | Function, value?: any): any;
+        filter(fn: Function, thisArg?: Object): Collection<key, value>;
+        find(propOrFn: string | Function, value?: any): value;
```

Comment:
That's uhh.. Not how filter works. You are correct in that using `set` can potentially introduce items into the collection that don't match the `value` Type, but I'd call that dev-error and Typescript would yell at you for attempting it. It'll compile, but it won't be pretty.

A filter function is intended to introduce a boolean value that will determine if a collection item makes it into the filtered collection. The way you were using is simply introducing a truthy value that will push all items into the filtered collection, resulting in no change.